speech-input.js

```diff
@@ -25,8 +25,8 @@
 		var parent = inputEl.parentNode;
 		var inputHeight = inputEl.offsetHeight;
 		var inputRightBorder = parseInt(getComputedStyle(inputEl).borderRightWidth, 10);
-		var buttonSize = 0.8 * inputHeight;
-		var buttonTop = 0.1 * inputHeight;
+		var buttonSize = 0.8 * (inputEl.dataset.buttonsize || inputHeight);
+		var buttonTop = 0.1 * (inputEl.dataset.buttonsize || inputHeight);
 
 		// max size for textareas
```

Comment:
With this setup, if you set a large `data-buttonsize` value on a `<textarea>`, it will be overridden by this "max size for textareas" JS code that follows. Is this what you're wanting?